frame/composable-traits/src/staking/mod.rs

```diff
@@ -102,7 +109,58 @@ pub struct RewardConfig<AssetId, Balance> {
 
 	/// The rewarding rate that increases the pool `total_reward`
 	/// at a given time.
-	pub reward_rate: Perbill,
+	pub reward_rate: RewardRate<Balance>,
+}
+
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo)]
+pub struct RewardRate<Balance> {
+	/// The period that the rewards are handed out in, in seconds.
+	// REVIEW(benluelo): NonZeroU64?
+	pub period: u64,
+	/// The amount that is rewarded each period.
+	pub amount: Balance,
+}
+
+pub struct BoundedU64<const MIN: u64 = 0, const MAX: u64 = { u64::MAX }> {
```

Comment:
Good one, but will the use of this cause any errors that are hard to understand?